aten/src/ATen/native/cuda/ForeachUnaryOp.cu

```diff
@@ -16,304 +15,189 @@ std::vector<Tensor> foreach_unary_op_complex(TensorList tensors) {
-    __device__ T operator()(T t) const { return std::NAME(t); }         \
-};                                                                      \
-                                                                        \
-std::vector<Tensor> foreach_tensor_##NAME##_cuda(TensorList tensors) {  \
-    check_foreach_api_restrictions(tensors);                            \
-                                                                        \
-    if (!can_use_fast_route(tensors)) {                                 \
-        return at::native::foreach_tensor_##NAME##_slow(tensors);       \
-    }                                                                   \
-                                                                        \
-    return foreach_unary_op_bfloat16<NAME1>(tensors);                   \
-}                                                                       \
-                                                                        \
-void foreach_tensor_##NAME##_cuda_(TensorList tensors) {                \
-    check_foreach_api_restrictions(tensors);                            \
-                                                                        \
-    if (!can_use_fast_route(tensors)) {                                 \
-        return at::native::foreach_tensor_##NAME##_slow_(tensors);      \
-    }                                                                   \
-                                                                        \
-    foreach_unary_op_bfloat16_<NAME1>(tensors);                         \
+std::vector<Tensor> foreach_tensor_sigmoid_cuda(TensorList tensors) {
+    check_foreach_api_restrictions(tensors);
+    if (!can_use_fast_route(tensors)) {
+        at::native::foreach_tensor_sigmoid_slow(tensors);
+    }
+    return floating_half_bfloat16<Sigmoid>(tensors);
 }
 
-FOREACH_UNARY_OP(ceil, Ceil);
-FOREACH_UNARY_OP(erfc, Erfc);
-FOREACH_UNARY_OP(expm1, Expm1);
-FOREACH_UNARY_OP(floor, Floor);
-FOREACH_UNARY_OP(lgamma, Lgamma);
-
-FOREACH_UNARY_OP_BFLOAT16(log1p, Log1p);
-FOREACH_UNARY_OP_BFLOAT16(erf, Erf);
-
-FOREACH_UNARY_OP_COMPLEX(acos, Acos);
-FOREACH_UNARY_OP_COMPLEX(asin, Asin);
-FOREACH_UNARY_OP_COMPLEX(atan, Atan);
-FOREACH_UNARY_OP_COMPLEX(cosh, Cosh);
-FOREACH_UNARY_OP_COMPLEX(tan, Tan);
-FOREACH_UNARY_OP_COMPLEX(sin, Sin);
-FOREACH_UNARY_OP_COMPLEX(sinh, Sinh);
-
-FOREACH_UNARY_OP_COMPLEX_BFLOAT16(exp, Exp);
-FOREACH_UNARY_OP_COMPLEX_BFLOAT16(sqrt, Sqrt);
-FOREACH_UNARY_OP_COMPLEX_BFLOAT16(cos, Cos);
-FOREACH_UNARY_OP_COMPLEX_BFLOAT16(tanh, Tanh);
-FOREACH_UNARY_OP_COMPLEX_BFLOAT16(log, Log);
-FOREACH_UNARY_OP_COMPLEX_BFLOAT16(log10, Log10);
-FOREACH_UNARY_OP_COMPLEX_BFLOAT16(log2, Log2);
+void foreach_tensor_sigmoid_cuda_(TensorList tensors) {
+    check_foreach_api_restrictions(tensors);
+    if (!can_use_fast_route(tensors)) {
+        at::native::foreach_tensor_sigmoid_slow_(tensors);
+    }
+
+    floating_half_bfloat16_<Sigmoid>(tensors);
+}
```

Comment:
Can this just be written as:
`OP_CUSTOM_FUNCTOR(floating_half_bfloat16, sigmoid, Sigmoid)` ?

Reply:
Updated